### PR TITLE
batches: show actions menu with preview + retry options when spec completes with errors

### PR DESCRIPTION
**Pull request by courier-new**
Closes https://github.com/sourcegraph/sourcegraph/issues/37084.

- Makes `viewerCanRetry` false if there's nothing to retry, aka if the batch spec completed successfully.
- Improves tooltip text for batch spec state badge.
- Prevents text wrap on Actions menu.
- Shows Actions menu instead of Preview button with warning when batch spec completes but with errors. The actions menu has options to "Preview with errors", "Retry failed workspaces", and "Edit spec". The preview option takes you to Step 4 as before.

Demo:

https://user-images.githubusercontent.com/8942601/173707813-88955d5a-24d2-4a54-b962-78cc2c08d87a.mov

## Test plan

Manually tested behavior, see video above. Component has storybook story coverage, which I also validated.

### CODE REVIEW

File: client/web/src/enterprise/batches/batch-spec/execute/BatchSpecStateBadge.tsx [hunk 1]
```diff
@@ -12,19 +12,22 @@ export interface BatchSpecStateBadgeProps {
 const getProps = (state: BatchSpecState): [variant: BadgeVariantType, tooltip: string] => {
     switch (state) {
         case BatchSpecState.PENDING:
-            return ['secondary', 'Not started']
+            return ['secondary', 'Execution has not been started.']
         case BatchSpecState.QUEUED:
-            return ['secondary', 'Waiting for executor']
+            return ['secondary', 'Waiting for next available executor.']
```
Comment (LawnGnome) [inline client/web/src/enterprise/batches/batch-spec/execute/BatchSpecStateBadge.tsx:17]:
```suggestion
            return ['secondary', 'Waiting for the next available executor.']
```

File: client/web/src/enterprise/batches/batch-spec/execute/ExecuteBatchSpecPage.story.tsx [hunk 2]
```diff
@@ -138,33 +137,6 @@ add('executing', () => (
...
-            {props => (
-                <MockedTestProvider link={new WildcardMockLink(FAILED_MOCKS)}>
-                    <ExecuteBatchSpecPage
-                        {...props}
-                        batchSpecID="spec1234"
-                        batchChange={{ name: 'my-batch-change', namespace: 'user1234' }}
-                        testContextState={{
-                            errors: {
-                                execute:
-                                    "Oh no something went wrong. This is a longer error message to demonstrate how this might take up a decent portion of screen real estate but hopefully it's still helpful information so it's worth the cost. Here's a long error message with some bullets:\n  * This is a bullet\n  * This is another bullet\n  * This is a third bullet and it's also the most important one so it's longer than all the others wow look at that.",
```
Comment (LawnGnome) [inline client/web/src/enterprise/batches/batch-spec/execute/ExecuteBatchSpecPage.story.tsx:156]: I don't know if we can lose this important content.

**Review by LawnGnome — APPROVED** [review]

File: client/web/src/enterprise/batches/batch-spec/execute/ExecuteBatchSpecPage.story.tsx (hunk 2 above)
Comment (courier-new) [reply client/web/src/enterprise/batches/batch-spec/execute/ExecuteBatchSpecPage.story.tsx:156]: You're absolutely right, I'll make sure it's preserved in the mock data. 😉

**Comment by adeola-ak** [comment]
looks so nice and clean!

**Pull request merged**